ext/pgsql/pgsql.c

```diff
@@ -5590,11 +5590,16 @@ PHP_PGSQL_API int php_pgsql_convert(PGconn *pg_link, const char *table_name, con
 						}
 						else {
 							/* FIXME: better regex must be used */
-							if (php_pgsql_convert_match(Z_STRVAL_PP(val), Z_STRLEN_PP(val), "^([+-]{0,1}[0-9]+)|([+-]{0,1}[0-9]*[\\.][0-9]+)|([+-]{0,1}[0-9]+[\\.][0-9]*)$", 0 TSRMLS_CC) == FAILURE) {
+							if (php_pgsql_convert_match(Z_STRVAL_PP(val), Z_STRLEN_PP(val), "^([+-]{0,1}[0-9]+)|([+-]{0,1}[0-9]*[\\.][0-9]+)|([+-]{0,1}[0-9]+[\\.][0-9]*)|([+-]{0,1}(inf)(inity){0,1})$", 1 TSRMLS_CC) == FAILURE) {
 								err = 1;
 							}
 							else {
+
 								ZVAL_STRING(new_val, Z_STRVAL_PP(val), 1);
+								if(php_pgsql_convert_match(Z_STRVAL_PP(val), Z_STRLEN_PP(val), "^([+-]{0,1}(inf)(inity){0,1})$", 1 TSRMLS_CC) == SUCCESS) {
```

Comment:
Yes, its clear.. but now doing some tests i see that using this approachs (check with php_pgsql_convert_match to only add quotes on 'infinity' values) generates notices when the regex not match.

```C
subs = (regmatch_t *)ecalloc(sizeof(regmatch_t), re.re_nsub+1);

	regerr = regexec(&re, str, re.re_nsub+1, subs, 0);
	if (regerr == REG_NOMATCH) {
#ifdef PHP_DEBUG
		php_error_docref(NULL TSRMLS_CC, E_NOTICE, "'%s' does not match with '%s'", str, regex);
#endif
		ret = FAILURE;
	}
```

Test result:
```
Notice: pg_update(): 'inf' does not match with '^([+-]{0,1}[0-9]+)|([+-]{0,1}[0-9]*[\.][0-9]+)|([+-]{0,1}[0-9]+[\.][0-9]*)$' in /opt/php/test_pgsql.php on line 19
string(52) "UPDATE "test_68638" SET "value"=E'inf' WHERE "id"=1;"

Notice: pg_update(): 'inf' does not match with '^([+-]{0,1}[0-9]+)|([+-]{0,1}[0-9]*[\.][0-9]+)|([+-]{0,1}[0-9]+[\.][0-9]*)$' in /opt/php/test_pgsql.php on line 21

Notice: pg_update(): '-inf' does not match with '^([+-]{0,1}[0-9]+)|([+-]{0,1}[0-9]*[\.][0-9]+)|([+-]{0,1}[0-9]+[\.][0-9]*)$' in /opt/php/test_pgsql.php on line 22

Notice: pg_update(): '+inf' does not match with '^([+-]{0,1}[0-9]+)|([+-]{0,1}[0-9]*[\.][0-9]+)|([+-]{0,1}[0-9]+[\.][0-9]*)$' in /opt/php/test_pgsql.php on line 23
```

I think one possible solutions to avoid this notice is to fork the "php_pgsql_convert_match" to "php_pgsql_match" that not generate this notice and use this function on the first "if" to not generate notice.

Like this:
```C
if (php_pgsql_match() == FAILURE){
      if (php_pgsql_convert_match("infinite") == SUCCESS) {
        add_quotes
     } else {
        err = 1;
     }
} else {
   //keep fast path un-touched
}
```